components/stack-exchange/new-question-for-keywords.js

```diff
@@ -0,0 +1,69 @@
+const stack_exchange = require('./stack-exchange.app');
+
+module.exports = {
+  name: "New Question for Specific Keywords",
+  description: "Emits an event when a new question is posted and related to a set of specific keywords",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    stack_exchange,
+    db: "$.service.db",
+    siteId: { propDefinition: [stack_exchange, "siteId"] },
+    keywords: { propDefinition: [stack_exchange, "keywords"] },
+    timer: {
+      type: '$.interface.timer',
+      default: {
+        intervalSeconds: 60 * 15, // 15 minutes
+      },
+    },
+  },
+  hooks: {
+    async activate() {
+      const fromDate = this._getCurrentEpoch();
+      this.db.set("fromDate", fromDate);
+    },
+  },
+  methods: {
+    _getCurrentEpoch() {
+      // The StackExchange API works with Unix epochs in seconds.
+      const epochInMs = +new Date();
+      return Math.floor(epochInMs / 1000);
+    },
+    generateMeta(data) {
+      const {
+        question_id: id,
+        creation_date: ts,
+        title,
+      } = data;
+      const summary = `New question: ${title}`;
+      return {
+        id,
+        summary,
+        ts,
+      };
+    },
+  },
+  async run() {
+    const fromDate = this.db.get("fromDate");
+    const toDate = this._getCurrentEpoch();
+    const keywordsQuery = this.keywords.join(',');
+    const searchParams = {
+      fromDate,
+      toDate,
+      order: 'asc',
+      sort: 'creation',
+      accepted: true,
+      closed: false,
+      site: this.siteId,
+      q: keywordsQuery,
+    };
+
+    const items = this.stack_exchange.advancedSearch(searchParams);
```

Comment:
Do you not need to await this?

Reply:
It's not needed actually. If you take a look at the `for` loop below, I'm actually doing the `await` there with the `for await` construction (as [explained here](https://javascript.info/async-iterators-generators#async-generators-finally)).

I tried adding an `await` there to see if the error message for the unresolved promises went away, but it didn't :disappointed: 

As a reference, the message I'm getting is the following:
```
Pipedream detected that component was still running code. Make sure you await all promises. (Reason: Socket)
```
As discussed offline, this could be a false positive caused by some issues with the platform itself.